### PR TITLE
Frontend v1

### PR DESCRIPTION
**Pull request by kartikey-codes**
# Changes in Frontend 

## 1) Framework Update
- Changed the framework to **React.js**

## 2) Graph Visualization
- Implemented **Graphs** using **Sigma.js** for better visualization and understanding.

## 3) UI/UX Changes
- Made major changes in the **ui/ux** for the *pypirecom*.
- Updated the **logo**.

## 4) Hosting Update
- Hosted the frontend separately on **Vercel**.

## 5) Filter Functionality
- Added **filter functionality**.

## 6) Performance Optimization
- Optimized the code to ensure the site **loads faster**.

### CODE REVIEW

File: V1/Frontend/src/App.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import '../node_modules/bootstrap/dist/js/bootstrap.bundle';
+import '../node_modules/bootstrap/dist/js/bootstrap.bundle.min.js'
+
+function App() {
+  return (
+    <Router>
+      <div>
+        <Routes>
+          <Route exact path="/" element={<Home />} />
+          <Route exact path="/about" element={<About />} />
```
Comment (Animesh2210) [inline V1/Frontend/src/App.js:21]: change to Explore package

File: V1/Frontend/src/Assets/data.json
Comment (Animesh2210) [inline V1/Frontend/src/Assets/data.json:1]: Remove

File: V1/Frontend/src/screens/About.jsx
Comment (Animesh2210) [inline V1/Frontend/src/screens/About.jsx:1]: Change name to Explore_Graph

File: V1/Frontend/src/screens/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    if (searchTextParam) {
+      setSearchText(searchTextParam);
+      // fetchData();
+    }
+  }, [location.search]); 
+
+  const fetchData = async () => {
+    setIsLoading(true);
+    try {
+      const response = await fetch(`https://pypirecom-api.azurewebsites.net/search?Search_Text=${searchText}`);
```
Comment (Animesh2210) [inline V1/Frontend/src/screens/Home.jsx:32]: Api url to be added in constants file

**Review by Develop-Things — CHANGES_REQUESTED** [review]

File: V1/Frontend/src/screens/About.jsx
Comment (kartikey-codes) [reply V1/Frontend/src/screens/About.jsx:1]: renamed it to ExploreGraphs

File: V1/Frontend/src/components/InfiniteLogo.jsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import img1 from '../Assets/LOGO_CAROUSEL/IMG-20240323-WA0006.jpg';
```
[the remark this answers is not in the document]
Comment (kartikey-codes) [reply V1/Frontend/src/components/InfiniteLogo.jsx]: done

File: V1/Frontend/src/components/Footer.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+export default function Footer() {
+  return (
+    <div className="container">
+      <footer className="py-3 my-4">
+        <p className="text-center text-muted" style={{fontSize: '18px', fontFamily: 'Segoe UI, Helvetica Neue, Arial, sans-serif', color: 'black'}}>
+          Made with <span style={{color: 'red', fontSize: '24px'}}>&#10084;&#65039;</span><br />by 
+          <span>
+            <a href="https://www.linkedin.com/in/bioenable/" target="_blank" rel="noopener noreferrer" className="mx-1" style={{color: '#0077b5'}}>Dr. Shyam Sundaram</a>, 
+            <a href="https://www.linkedin.com/in/animesh2210/" target="_blank" rel="noopener noreferrer" className="mx-1" style={{color: '#0077b5'}}>Animesh Verma</a>, and 
+            <a href="https://www.linkedin.com/in/sridhar-aluru-8b9904176/" target="_blank" rel="noopener noreferrer" className="mx-1" style={{color: '#0077b5'}}>Sridhar Aluru</a>
```
[the remark this answers is not in the document]
Comment (kartikey-codes) [reply V1/Frontend/src/components/Footer.jsx]: done

File: V1/Frontend/src/Assets/data.json
Comment (kartikey-codes) [reply V1/Frontend/src/Assets/data.json:1]: done removed

File: V1/Frontend/src/screens/Home.jsx (hunk 2 above)
Comment (kartikey-codes) [reply V1/Frontend/src/screens/Home.jsx]: done created a js file to store constant value for the BASE URL

File: V1/Frontend/src/api-endpoint.js
Comment (Animesh2210) [inline V1/Frontend/src/api-endpoint.js:1]: add local url also and make sure  it works on local with backend also not only with hosted api

**Review by Develop-Things — APPROVED** [review]
Will go through graph.jsx file and home.jsx and explorepacakage.jsx again. But approving now

**Pull request merged**